### PR TITLE
Show CES survey after editing an order

### LINKED ISSUE #5347: Customer effort score - Trigger on edit order
Part of #5259

Trigger the customer effort score modal when editing an order.

Also include the store's current order count in the tracks data (as `order_count`).

Copy for the CES modal should be `How easy was it to update an order?`

### PR DESCRIPTION
**Pull request by mattsherman**
Fixes #5347 

This PR triggers the display of the Customer Effort Score survey modal after editing an order. It only displays if tracking is enabled and the survey has not already been shown.

### Detailed test instructions:

Note: Depending on other outstanding PRs, you may need to make sure that...

- Tracking is enabled: `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com`
- The options tracking whether the survey has been shown is cleared out: `delete from wp_options where option_name = 'woocommerce_ces_shown_for_actions'` executed

Steps:

- Enable the debugging of Tracks events in your browser console: `localStorage.setItem( 'debug', 'wc-admin:tracks' );`
- Add an order (checkout from cart) on store frontend of site.
- Click on an order in wp-admin.
- Verify that the CES survey snackbar is not shown.
- Go back to the list of orders.
- Bulk modify some orders.
- Click on an order in wp-admin.
- Verify that the CES survey snackbar is not shown.
- Go back to the list of orders.
- Edit an order in wp-admin.
- Verify the CES survey snackbar is displayed.
- Click "Provide Feedback" in the snackbar.
- Verify that the "How easy was it to update an order?" survey modal is shown (how this modal looks will depend on what other PRs related to this have been merged; this PR is not concerned with how the modal looks)
- Verify that if you complete the survey (again, how this is done depends on the state of other open PRs), a Tracks event for `wcadmin_ces_feedback` is logged, with:
    - `action` set to `shop_order_update` 
    - `order_count` set to the number of orders in the store

Other things to test:

- Make sure that the CES modal is shown as expected when adding and editing Products, according to the test instructions in #5352

### Changelog Note:

No changelog note required, as this is part of the larger Customer Effort Score feature being added.

### CODE REVIEW

**Review by adrianduffell — DISMISSED** [review]
@mattsherman Thanks for jumping on this! Overall it looks good to me.

I found some cases where `enqueue_ces_survey_for_edited_shop_order` is called unexpectedly.

- When a customer places an order.
- When orders are bulk deleted.

In both these cases, the snackbar is visible the next time an admin edits any order.

Could a check be made to ensure the order was updated from the admin page?

**Comment by mattsherman** [comment]
Thanks for the review @adrianduffell -- I'll add a check to ensure the order was updated from the admin page, take care of the merge conflict, and then put back up for review!

**Comment by mattsherman** [comment]
@adrianduffell I have made changes to this PR to ensure that only edits from the Orders edit page trigger the CES modal. Orders created/modified from the front-end of the store or through bulk actions should not trigger the CES modal.

One of the changes I made also will affect other uses of CES triggering (I updated the PR test details to reflect this), so it would be good to test adding/editing of Products as well.

File: src/Features/CustomerEffortScoreTracks.php [hunk 1]
```diff
@@ -114,10 +133,27 @@ public function run_on_transition_post_status(
 		$old_status,
 		$post
 	) {
-		if ( 'product' !== $post->post_type ) {
+		if ( ! in_array( $post->post_type, self::TRACKED_POST_TYPES, true ) ) {
```
Comment (joshuatf) [inline src/Features/CustomerEffortScoreTracks.php:136]: This `if` clause seems a little redundant with the below conditional.

File: src/Features/CustomerEffortScoreTracks.php [hunk 2]
```diff
@@ -129,6 +165,19 @@ public function run_on_transition_post_status(
 		}
 	}
 
+	/**
+	 * Maybe enqueue the CES survey, if shop order is being edited.
+	 *
+	 * @param string $new_status The new status.
+	 * @param string $old_status The old status.
+	 */
+	private function maybe_enqueue_ces_survey_for_shop_order(
```
Comment (joshuatf) [inline src/Features/CustomerEffortScoreTracks.php:174]: This function uses a `maybe_` prefix, but the logic and condition for this appear to be in `enqueue_ces_survey_for_edited_shop_order`.

File: src/Features/CustomerEffortScoreTracks.php [hunk 3]
```diff
@@ -78,15 +92,20 @@ private function enable_survey_enqueing_if_tracking_is_enabled() {
...
-				$this,
-				'run_on_transition_post_status',
-			),
-			10,
-			3
-		);
+		// Only hook up the transition_post_status action handler
+		// if on the edit page.
+		global $pagenow;
+		if ( 'post.php' === $pagenow ) {
```
Comment (joshuatf) [inline src/Features/CustomerEffortScoreTracks.php:98]: Nice workaround!

**Review by joshuatf — APPROVED** [review]
Left a couple nitpicky comments, but this is testing well and pre-approving!

Side note: the "Give Feedback" action of the modal does not produce any results for me.  Is this expected or a bug?

**Comment by mattsherman** [comment]
> Left a couple nitpicky comments, but this is testing well and pre-approving!
> 
> Side note: the "Give Feedback" action of the modal does not produce any results for me. Is this expected or a bug?

@joshuatf Thanks for testing and reviewing this! Yes, the Give Feedback action of the modal doesn't do anything currently... that modal is just a placeholder until we merge the actual modal (#5515).

File: src/Features/CustomerEffortScoreTracks.php (hunk 1 above)
Comment (mattsherman) [reply src/Features/CustomerEffortScoreTracks.php:136]: Agreed! I will remove this check before I merge.

File: src/Features/CustomerEffortScoreTracks.php (hunk 2 above)
Comment (mattsherman) [reply src/Features/CustomerEffortScoreTracks.php:174]: Seems like a remnant of an earlier version of the implementation. I will clean this up before merging.

**Comment by mattsherman** [comment]
Setting label to `[Status] Ready to Merge` to kick off checks prior to merging.

**Pull request merged**